Source/WebCore/dom/EventTarget.h

```diff
@@ -172,32 +165,33 @@ class EventTarget : public ScriptWrappable, public CanMakeWeakPtr<EventTarget, W
         IsNode = 1 << 1,
     };
 
-    void setHasEventTargetData(bool flag) const
-    {
-        uint16_t bitfield = weakPtrFactory().bitfield();
-        if (flag)
-            bitfield |= static_cast<uint16_t>(EventTargetFlag::HasEventTargetData);
-        else
-            bitfield &= ~static_cast<uint16_t>(EventTargetFlag::HasEventTargetData);
-        weakPtrFactory().setBitfield(bitfield);
-    }
-    
     EventTargetData& ensureEventTargetData()
     {
         if (auto* data = eventTargetData())
             return *data;
         initializeWeakPtrFactory();
         WTF::storeStoreFence();
-        setHasEventTargetData(true);
+        setEventTargetFlag(EventTargetFlag::HasEventTargetData, true);
         return weakPtrFactory().impl()->eventTargetData();
     }
 
     virtual void eventListenersDidChange() { }
 
+    bool hasEventTargetFlag(EventTargetFlag flag) const { return weakPtrFactory().bitfield() & static_cast<uint16_t>(flag); }
+    void setEventTargetFlag(EventTargetFlag flag, bool value)
```

Comment:
Given the number of lines, I think we should move it out of the class and mark it as inline instead of inlining it in the class.